### PR TITLE
LTD-5603-privilege-escalation-restrict-view-profile 

### PR DESCRIPTION
**Pull request by depsiatwal**
https://uktrade.atlassian.net/browse/LTD-5603

This restricts the view profile and removes the change links so only a super-user or someone from Admin team can change roles and teams. 

This PR is more about injecting in the permissions for following PR which will restrict  the change profile screen. 

At the moment i've decided to keep this as a lite form but the change user profile screen may be easier to convert to a crispy_form at that point I will consider moving both  views.

### CODE REVIEW

File: caseworker/core/rules.py [hunk 1]
```diff
@@ -180,6 +181,19 @@ def is_organisation_active(request, organisation):
...
 
 
+@rules.predicate
+def is_super_user(request):
+    user = get_logged_in_caseworker(request)
+    return user.get("role", {}).get("id") == SUPER_USER_ROLE_ID
+
+
+@rules.predicate
+def user_is_not_logged_in_user(request, user):
```
Comment (kevincarrogan) [inline caseworker/core/rules.py:191]: The naming of this rule had me confused. 

When I first read this I thought it was a check to see if the user was logged in and not that it was a check to see if the user passed in is the same as the currently logged in user.

File: caseworker/core/rules.py [hunk 2]
```diff
@@ -180,6 +181,19 @@ def is_organisation_active(request, organisation):
...
+    caseworker = get_logged_in_caseworker(request)
+    caseworker_user_id = caseworker["id"]
+    return True if not user else user.get("user", {}).get("id") != caseworker_user_id
```
Comment (kevincarrogan) [inline caseworker/core/rules.py:194]: I think this could be written in a way that’s much clearer to understand.

Comment (depsiatwal) [reply caseworker/core/rules.py]: updated

File: caseworker/core/rules.py (hunk 1 above)
Comment (depsiatwal) [reply caseworker/core/rules.py]: is that name clearer ?

**Review by kevincarrogan — APPROVED** [review]

File: caseworker/core/rules.py [hunk 3]
```diff
@@ -180,6 +181,21 @@ def is_organisation_active(request, organisation):
...
+    return user.get("role", {}).get("id") == SUPER_USER_ROLE_ID
+
+
+@rules.predicate
+def check_user_is_not_logged_in_caseworker(request, user):
+    caseworker = get_logged_in_caseworker(request)
+    caseworker_user_id = caseworker["id"]
+    if not user:
+        return True
+    return user.get("user", {}).get("id") != caseworker_user_id
```
Comment (currycoder) [inline caseworker/core/rules.py:196]: It seems like we are passing through a datastructure to this function that contains the user...  Could we not enforce that this function accepts user as `user` directly?

And do we really need `if not user: ...`?  Seems like we should always have a value and if not it's ok to hard error

**Review by currycoder — APPROVED** [review]
One suggestion, but otherwise looking good to me

File: caseworker/core/rules.py (hunk 3 above)
Comment (depsiatwal) [reply caseworker/core/rules.py:196]: ahh I merged too fast will update in my next follow up PR which also touches this area. 

what do you mean by user directly a structured user object ?

Comment (depsiatwal) [reply caseworker/core/rules.py:196]: oh do you mean pass in 
user["user'] as opposed to  data which is what this dump lite view is doing

Comment (currycoder) [reply caseworker/core/rules.py:196]: Yeah just pass in the user as `user` rather than whatever other cruft.  Happy for it to follow - no worries.

**Pull request merged**